src/groups/mqb/mqbnet/mqbnet_clusteractivenodemanager.cpp

```diff
@@ -244,11 +244,19 @@ ClusterActiveNodeManager::ClusterActiveNodeManager(
 , d_activeNodeIt(d_nodes.end())
 , d_useExtendedSelection(false)
 {
+    bool clusterNodeInLocalDC = true;
     for (mqbnet::Cluster::NodesList::const_iterator it = nodes.begin();
          it != nodes.end();
          ++it) {
         mqbnet::ClusterNode* node = *it;
         d_nodes[node].d_status    = bmqp_ctrlmsg::NodeStatus::E_UNAVAILABLE;
+        clusterNodeInLocalDC      = clusterNodeInLocalDC &&
+                               d_dataCenter == node->dataCenter();
```

Comment:
I am not sure if the logics here is correct.
The `clusterNodeInLocalDC ` flag is initially true, but it can only turn `false` in this loop. So if we have at least one node in another datacenter, the flag will turn `false` and we will do extended selection.

I would revert the flag here and do smth like
```
bool clusterNodeInLocalDC = false;
for (...) {
   ...
   clusterNodeInLocalDC = clusterNodeInLocalDC ||
                               d_dataCenter == node->dataCenter();
   // will turn `true` if at least one node has the same cluster
}
```